joomla/joomla-cms
# Fix bug clearing datepicker unintentionally

Pull Request for Issue #17741.


### Summary of Changes
Added `type="button"` to the datepicker's action buttons as the browser interprets buttons
without a type as submit button, so it (the first button in the form) gets triggered when pressing enter within the containing form.

See #17741.

### Testing Instructions

* Open an admin form with date field (i.e. an existing article)
* Go to the Publishing options
* Make sure there is a date value selected for Start Publishing and Created Date.
* If not set these date fields and save the article.
* Click into the input field of the Created Date option then.
* Press the enter key

### Expected result
Nothing should change with the selected dates visible in Start Publishing and in Created Date

### Actual result
Input of Start Publishing is cleared.
(It is always the input of the first date field in the form that is cleared)


### Documentation Changes Required
None.

### Notes
Maybe the form should be submitted when pressing enter, this isn't happening automatically as the toolbar buttons are outside of the form.

Developed @icampus
